### PR TITLE
#31 Add sliders to map prototype

### LINKED ISSUE #31: Prototype Sliders for Oblasts

### PR DESCRIPTION
**Pull request by jeffwklein**
Closes #31 

## ⚠️  TODO: get map to react to slider changes

Screen Shot 2023-02-06 at 7 49 03 PM
Screen Shot 2023-02-06 at 7 49 27 PM

### CODE REVIEW

File: pages/map.vue [hunk 1]
```diff
@@ -263,6 +271,13 @@ function updateMap(geographicData) {
       const oblastData = parsedDataByName.value[shapeName1];
       const shapeValue = oblastData ? oblastData[valueKey.value] : 0;
 
+      // TODO: show scaled values on map.
+      //  - parsedDataByName does not include the scaled value, it contains the original value.
+      //  - Either we need to get the scaled value from sortedDataByGrainType, or scale it right here using the formatAndScaleValue helper
+      // i.e.
+      // const scaledShapeValue = shapeValue ? formatAndScaleValue(shapeValue, normalizeOblastName(oblastData.oblastNameUkrainian)) : 0;
+      // console.log(shapeValue, scaledShapeValue);
```
Comment (jeffwklein) [inline pages/map.vue:279]: this is the final step of getting the map to react to the slider changes.

File: pages/map.vue [hunk 2]
```diff
@@ -115,7 +114,7 @@ function clearSelectedOblasts() {
   selectedOblasts.value = []
 }
 
-// HELLO 
+// HELLO
```
Comment (ejfox) [inline pages/map.vue:117]: HELLO

**Review by ejfox — APPROVED** [review]

File: pages/map.vue (hunk 2 above)
Comment (jeffwklein) [reply pages/map.vue:117]: HELLO

Comment (curran) [reply pages/map.vue:118]:
```suggestion
```

GOODBYE

File: pages/map.vue [hunk 3]
```diff
@@ -251,17 +273,56 @@ function initMap(geographicData) {
...
+  console.log("raw objects", geographicData.objects['stanford-pp624tm0074-geojson'])
+  // let geodata = topojson.feature(geographicData, geographicData.objects['stanford-pp624tm0074-geojson']).features
+  // Merge geometries so we end up with Oblast-level shapes.
+  const featureCollection = aggregate(
+    geographicData,
+    geographicData.objects['stanford-pp624tm0074-geojson'],
+    'name_1'
+  );
+
+  let map = d3.select(mapSvg.value)
```
Comment (curran) [inline pages/map.vue:285]:
```suggestion
  const map = d3.select(mapSvg.value)
```

File: pages/map.vue [hunk 4]
```diff
@@ -251,17 +273,56 @@ function initMap(geographicData) {
...
+  let paths = map.selectAll('path')
```
Comment (curran) [inline pages/map.vue:286]:
```suggestion
  const paths = map.selectAll('path')
```

When I see `let` I immediately start looking around to see where it's mutated. IMO `const` is simpler to comprehend if there are no reassignments.

**Review by curran — APPROVED** [review]
LGTM! Minor nits with commit suggestions.

**Comment by ejfox** [comment]
Merged @curran's commits- I really like that interface where it is one-click, very easy. Not sure if that takes extra work, but would love to learn how you do that (a short video??)

**Pull request merged**